libs/comments/api/src/lib/comment.service.ts

```diff
@@ -0,0 +1,360 @@
+import {Injectable} from '@nestjs/common'
+import {Comment, CommentAuthorType, CommentsRevisions, PrismaClient} from '@prisma/client'
+import {SortOrder} from '@wepublish/utils/api'
+import {
+  CalculatedRating,
+  CommentRatingSystemAnswer,
+  CommentState,
+  PublicCommentSort,
+  RatingSystemType
+} from './comment.model'
+import {RatingSystemService} from './rating-system'
+import {
+  AnonymousCommentError,
+  AnonymousCommentRatingDisabledError,
+  AnonymousCommentsDisabledError,
+  ChallengeMissingCommentError,
+  CommentAuthenticationError,
+  CommentLengthError,
+  countRichtextChars,
+  hasPermission,
+  InvalidStarRatingValueError,
+  NotAuthorisedError,
+  NotFound,
+  UserInputError,
+  UserSession
+} from '@wepublish/api'
+import {CanCreateApprovedComment} from '@wepublish/permissions'
+import {ChallengeService} from '@wepublish/challenge/api'
+import {CommentInput, CommentUpdateInput} from './comment.input'
+import {SettingName, SettingsService} from '@wepublish/settings/api'
+import {CommentWithTags} from './comment.types'
+
+export interface CommentWithRevisions {
+  title: string | null
+  lead: string | null
+  text: string | null
+
+  [key: string]: any
+}
+
+@Injectable()
+export class CommentService {
+  constructor(
+    private readonly prisma: PrismaClient,
+    private readonly ratingSystem: RatingSystemService,
+    private readonly settingsService: SettingsService,
+    private readonly challengeService: ChallengeService
+  ) {}
+
+  async getPublicChildrenCommentsByParentId(parentId: string, userId: string | null) {
+    const comments = await this.prisma.comment.findMany({
+      where: {
+        AND: [
+          {parentID: parentId},
+          {OR: [userId ? {userID: userId} : {}, {state: CommentState.approved}]}
```

Comment:
This now doesn't return unapproved comments anymore for admins. Use preview mode.